src/PhpseclibV3/SftpAdapter.php

```diff
@@ -274,6 +274,10 @@ public function listContents(string $path, bool $deep): iterable
         $location = $this->prefixer->prefixPath(rtrim($path, '/')) . '/';
         $listing = $connection->rawlist($location, false);
 
+        if (false === $listing) {
+            return [];
```

Comment:
```suggestion
            return;
```

seems better since we are returning a `\Generator`?

Reply:
agreed